crates/intrinsic-test/src/common/intrinsic_helpers.rs

```diff
@@ -12,23 +13,28 @@ use super::values::value_for_array;
 pub enum TypeKind {
     BFloat,
     Float,
-    Int,
-    UInt,
+
+    // if signed, then the inner value is true
+    Int(bool),
```

Comment:
Rather than using a `bool`, considering instead something like:
```rust
enum Signed {
    Signed,
    Unsigned,
}
```

This ends up making the code much more readable.